name.abuchen.portfolio/src/name/abuchen/portfolio/model/ClientFactory.java

```diff
@@ -838,6 +842,16 @@ else if (flags.contains(SaveFlag.COMPRESSED))
         }
     }
 
+    private static void removeWronglyAddedSecurities(Client client)
+    {
+        client.getWatchlists().forEach(w -> new ArrayList<>(w.getSecurities()).forEach(s -> {
+            if (!client.getSecurities().contains(s))
```

Comment:
I'm not sure if "contains" should be replaced by "equals"?! 

maybe with "contains"...
```
List<Watchlist> watchlists = client.getWatchlists();
List<Security> clientSecurities = client.getSecurities();

for (Watchlist watchlist : watchlists) {
    List<Security> watchlistSecurities = new ArrayList<>(watchlist.getSecurities());
    for (Security security : watchlistSecurities) {
        if (!clientSecurities.contains(security)) {
            watchlist.getSecurities().remove(security);
        }
    }
}
```

Reply:
Isn't this saying the same thing - just with more code? I am not sure I understand your comment

Reply:
Die iterierung in der Watchlist hat Vorrang...?!

Reply:
Ah okay, I believe this ```w -> new ArrayList<>(w.getSecurities()``` also copies the securities to be able to modify the underlying list.

Reply:
Yes, I think it's the same just with ``for``-statement instead of ``.forEach``. The only difference I can see is that @Nirus2000 made only one single call to ``client.getSecurities()``. In my proposal this fcuntion is called N times, were N is the sum of all securities in all the watchlists.

Reply:
Allerdings sollten sichergestellt werden, dass die Security-Klasse eine korrekte equals()- und UUID-Methode implementiert ist, damit "contains()" das gewünschten Ergebnisse liefert. 
Wenn das nicht der Fall ist, müssen die contains()-Logik anpassen werden, um die Wertpapiere besser zu vergleichen.
Das habe ich jetzt nicht prüfen können... war von mir nur ein Querschuss... I dont know

Reply:
> Yes, I think it's the same just with `for`-statement instead of `.forEach`. The only difference I can see is that @Nirus2000 made only one single call to `client.getSecurities()`. In my proposal this fcuntion is called N times, were N is the sum of all securities in all the watchlists.

Ja, genau das ist der Unterschied, welche aber wichtig ist.

Reply:
Egal... wird schon schief gehen... :rofl::rofl::rofl::rofl::rofl::rofl::rofl::rofl::rofl:

Reply:
```Security``` never implemented ```equals``` as we want to work with object identity. I would think Client#getSecurities will be inlined by the JIT (it will run once per file - I do not think it will be a problem).

Anyway, I tested it with a sample, it works as far as I can tell, the release is already for notarization at Apple. If it fails, it will be one me. :-)

Thanks @OnkelDok for the pull request, thanks @Nirus2000 for looking over it

Reply:
I tested it with corrupt file (watchlist that contains wrongly added securities and also valid securities) and only the wrongly added securities were deleted. For this functionality I did not tried to optimize the performance, because this function normally is only called once per file and the ``client.getSecurities()`` is not that expensive operation.